src/routes/webhooks/customerio.ts

```diff
@@ -37,62 +38,133 @@ type MarketingCtaPayload = {
+      );
 
-  fastify.post<MarketingCtaPayload>('/marketing_cta', {
-    config: {
-      rawBody: true,
-    },
-    handler: async (req, res) => {
-      try {
-        const { userId, marketingCtaId } = req.body;
-
-        const con = await createOrGetConnection();
-        await con.getRepository(UserMarketingCta).upsert(
-          {
-            userId: userId,
-            marketingCtaId: marketingCtaId,
-          },
-          ['userId', 'marketingCtaId'],
-        );
-
-        await cachePrefillMarketingCta(con, userId);
-
-        return res.send({ success: true });
-      } catch (err) {
-        logger.error({ err }, 'Error processing CIO webhook');
-        return res.status(400).send({ success: false });
-      }
+      fastify.post<MarketingCtaPayload>('/', {
+        config: {
+          rawBody: true,
+        },
+        handler: async (req, res) => {
+          try {
+            const { userId, marketingCtaId } = req.body;
+
+            const con = await createOrGetConnection();
+            await con.getRepository(UserMarketingCta).upsert(
+              {
+                userId: userId,
+                marketingCtaId: marketingCtaId,
+              },
+              ['userId', 'marketingCtaId'],
+            );
+
+            await cachePrefillMarketingCta(con, userId);
+
+            return res.send({ success: true });
+          } catch (err) {
+            logger.error({ err }, 'Error processing CIO webhook');
+            return res.status(400).send({ success: false });
+          }
+        },
+      });
+
+      fastify.delete<MarketingCtaPayload>('/', {
+        config: {
+          rawBody: true,
+        },
+        handler: async (req, res) => {
+          try {
+            const { userId, marketingCtaId } = req.body;
+
+            const con = await createOrGetConnection();
+            await con.getRepository(UserMarketingCta).delete({
+              userId: userId,
+              marketingCtaId: marketingCtaId,
+            });
+
+            await cachePrefillMarketingCta(con, userId);
+
+            return res.send({ success: true });
+          } catch (err) {
+            logger.error({ err }, 'Error processing CIO webhook');
+            return res.status(400).send({ success: false });
+          }
+        },
+      });
     },
-  });
+    { prefix: '/marketing_cta' },
+  );
 
-  fastify.delete<MarketingCtaPayload>('/marketing_cta', {
+  fastify.post<ReportingEvent>('/reporting', {
     config: {
       rawBody: true,
     },
     handler: async (req, res) => {
-      try {
-        const { userId, marketingCtaId } = req.body;
-
-        const con = await createOrGetConnection();
-        await con.getRepository(UserMarketingCta).delete({
-          userId: userId,
-          marketingCtaId: marketingCtaId,
-        });
+      const valid = verifyCIOSignature(
+        process.env.CIO_REPORTING_WEBHOOK_SECRET,
+        req,
+      );
+      if (!valid) {
+        req.log.warn('invalid signature');
```

Comment:
Either put the log as debug, or give it more context than `invalid signature`?

Reply:
done. good point